File: searching/binarySearch.java
package searching;


public class binarySearch {
    static int search(int[] arr,int k )
    {
//        Arrays.sort(arr);
        int l = 0, m;
        int h = arr.length-1;
        while (l<=h)
        {
            m = l+(h-l)/2;
            if (arr[m] == k)
            {
                return  m   ;
            }
            if (arr[m] > k)
            {
                h = m-1;
            }
            else
            {
                l = m+1;
            }
        }
        return -1;
    }

    public static void main(String[] args) {

        int [] arr = {-1,0,3,5,9,12};
        // int n = arr.length;
        int target = 2;

        int s = search(arr,target);
        if (s == -1)
        {
            System.out.println(target+ " is not present in the array");
        }
        else
        {
            System.out.println(target+ " is present in the array at the index " +s);
        }


    }
}

File: searching/binarySearch.cpp
/*binary search iterative
time - O(logn)*/
#include<bits/stdc++.h>
#include<iostream>
using namespace std;

int bSearch(int arr[],int n, int target)
{
    int i = 0, j = n;
    while (i<=j)
    {
        int m = (i+j)/2;
        if (arr[m] == target)
        {
            return m;
        }
        if (arr[m] > target)
        {
            j = m-1;
        }
        else
        {
            i = m+1;
        }   
    }

    return -1;

}

int main()
{
    int arr[] = {43,56,67,68,76,89,90};
    int n = sizeof(arr)/sizeof(arr[0]);
    cout << bSearch(arr,n,56);
    
    return 0;
}